programs/server/Server.cpp

```diff
@@ -2111,13 +2115,24 @@ void Server::updateServers(
 
     std::erase_if(servers, std::bind_front(check_server, " (from one of previous reload)"));
 
+    Poco::Util::AbstractConfiguration & previous_config = latest_config ? *latest_config : this->config();
+
     for (auto & server : servers)
     {
         if (!server.isStopping())
         {
             bool has_host = std::find(listen_hosts.begin(), listen_hosts.end(), server.getListenHost()) != listen_hosts.end();
             bool has_port = !config.getString(server.getPortName(), "").empty();
-            if (!has_host || !has_port || config.getInt(server.getPortName()) != server.portNumber())
+
+            /// NOTE: better to compare using getPortName() over use
+            /// dynamic_cast<> since HTTPServer is also used for prometheus and
```

Comment:
```suggestion
            /// NOTE: better to compare using getPortName() over using
            /// dynamic_cast<> since HTTPServer is also used for prometheus and
```